Kubernetes-maven-plugin doesn't work with Apple M1 CPU

### Description

The `kubernetes-maven-plugin` doesn't work on Apple M1 machines. 
Similar issue was also reported in `docker-maven-plugin` and fixed by upgrading `jnr-unixsocket` to at least [0.38.17](https://github.com/jnr/jnr-unixsocket/issues/95) 

### AI
- [x] CQ for the new version 0.38.17 of `com.github.jnr:jnr-unixsocket` https://dev.eclipse.org/ipzilla/show_bug.cgi?id=24105
- [x] ~~CQ for the (overridden?) version 5.0.3 of  `org.ow2.asm:asm` https://dev.eclipse.org/ipzilla/show_bug.cgi?id=24106~~ Covered by [Eclipse Dash](https://www.eclipse.org/projects/handbook/#ip-dash-db) dependency report
- [x] The fix: https://github.com/eclipse/jkube/pull/1471

### Info

* Eclipse JKube version : 1.6.0, 1.7.0, 1.8.0-SNAPSHOT
* Maven version (`mvn -v`) : 
```
Apache Maven 3.8.5 (3599d3414f046de2324203b78ddcf9b5e4388aa0)
Maven home: /opt/homebrew/Cellar/maven/3.8.5/libexec
Java version: 18, vendor: Homebrew, runtime: /opt/homebrew/Cellar/openjdk/18/libexec/openjdk.jdk/Contents/Home
Default locale: en_GB, platform encoding: UTF-8
OS name: "mac os x", version: "12.3.1", arch: "aarch64", family: "mac"
```

### How to reproduce  (macOS with M1 CPU only): 
```bash
 cd jkube/quickstarts/maven/hello-world 
 mvn clean install
 mvn k8s:build 
```
<details><summary>maven output:</summary>
<pre>
 ~/Development/jkube/quickstarts/maven/hello-world   master ±  mvn k8s:build
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------< org.eclipse.jkube.quickstarts.maven:helloworld >-----------
[INFO] Building Eclipse JKube :: Quickstarts :: Maven :: Hello World 1.7.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- kubernetes-maven-plugin:1.7.0:build (default-cli) @ helloworld ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.580 s
[INFO] Finished at: 2022-04-23T21:05:39+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.jkube:kubernetes-maven-plugin:1.7.0:build (default-cli) on project helloworld: Execution default-cli of goal org.eclipse.jkube:kubernetes-maven-plugin:1.7.0:build failed: An API incompatibility was encountered while executing org.eclipse.jkube:kubernetes-maven-plugin:1.7.0:build: java.lang.UnsatisfiedLinkError: could not load FFI provider jnr.ffi.provider.jffi.Provider
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.eclipse.jkube:kubernetes-maven-plugin:1.7.0
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/kubernetes-maven-plugin/1.7.0/kubernetes-maven-plugin-1.7.0.jar
[ERROR] urls[1] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-config-service/1.7.0/jkube-kit-config-service-1.7.0.jar
[ERROR] urls[2] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-config-resource/1.7.0/jkube-kit-config-resource-1.7.0.jar
[ERROR] urls[3] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-build-service-docker/1.7.0/jkube-kit-build-service-docker-1.7.0.jar
[ERROR] urls[4] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-build-api/1.7.0/jkube-kit-build-api-1.7.0.jar
[ERROR] urls[5] = file:/Users/anowak/.m2/repository/commons-codec/commons-codec/1.15/commons-codec-1.15.jar
[ERROR] urls[6] = file:/Users/anowak/.m2/repository/com/github/jnr/jnr-unixsocket/0.12/jnr-unixsocket-0.12.jar
[ERROR] urls[7] = file:/Users/anowak/.m2/repository/com/github/jnr/jnr-ffi/2.0.9/jnr-ffi-2.0.9.jar
[ERROR] urls[8] = file:/Users/anowak/.m2/repository/com/github/jnr/jffi/1.2.11/jffi-1.2.11.jar
[ERROR] urls[9] = file:/Users/anowak/.m2/repository/com/github/jnr/jffi/1.2.11/jffi-1.2.11-native.jar
[ERROR] urls[10] = file:/Users/anowak/.m2/repository/org/ow2/asm/asm/5.0.3/asm-5.0.3.jar
[ERROR] urls[11] = file:/Users/anowak/.m2/repository/org/ow2/asm/asm-commons/5.0.3/asm-commons-5.0.3.jar
[ERROR] urls[12] = file:/Users/anowak/.m2/repository/org/ow2/asm/asm-analysis/5.0.3/asm-analysis-5.0.3.jar
[ERROR] urls[13] = file:/Users/anowak/.m2/repository/org/ow2/asm/asm-tree/5.0.3/asm-tree-5.0.3.jar
[ERROR] urls[14] = file:/Users/anowak/.m2/repository/org/ow2/asm/asm-util/5.0.3/asm-util-5.0.3.jar
[ERROR] urls[15] = file:/Users/anowak/.m2/repository/com/github/jnr/jnr-x86asm/1.0.2/jnr-x86asm-1.0.2.jar
[ERROR] urls[16] = file:/Users/anowak/.m2/repository/com/github/jnr/jnr-constants/0.9.1/jnr-constants-0.9.1.jar
[ERROR] urls[17] = file:/Users/anowak/.m2/repository/com/github/jnr/jnr-enxio/0.12/jnr-enxio-0.12.jar
[ERROR] urls[18] = file:/Users/anowak/.m2/repository/com/github/jnr/jnr-posix/3.0.29/jnr-posix-3.0.29.jar
[ERROR] urls[19] = file:/Users/anowak/.m2/repository/org/bouncycastle/bcpkix-jdk15on/1.61/bcpkix-jdk15on-1.61.jar
[ERROR] urls[20] = file:/Users/anowak/.m2/repository/org/bouncycastle/bcprov-jdk15on/1.61/bcprov-jdk15on-1.61.jar
[ERROR] urls[21] = file:/Users/anowak/.m2/repository/org/apache/commons/commons-text/1.9/commons-text-1.9.jar
[ERROR] urls[22] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-build-service-jib/1.7.0/jkube-kit-build-service-jib-1.7.0.jar
[ERROR] urls[23] = file:/Users/anowak/.m2/repository/com/google/cloud/tools/jib-core/0.20.0/jib-core-0.20.0.jar
[ERROR] urls[24] = file:/Users/anowak/.m2/repository/com/google/cloud/tools/jib-build-plan/0.4.0/jib-build-plan-0.4.0.jar
[ERROR] urls[25] = file:/Users/anowak/.m2/repository/com/google/http-client/google-http-client/1.34.0/google-http-client-1.34.0.jar
[ERROR] urls[26] = file:/Users/anowak/.m2/repository/io/opencensus/opencensus-api/0.24.0/opencensus-api-0.24.0.jar
[ERROR] urls[27] = file:/Users/anowak/.m2/repository/io/grpc/grpc-context/1.22.1/grpc-context-1.22.1.jar
[ERROR] urls[28] = file:/Users/anowak/.m2/repository/io/opencensus/opencensus-contrib-http-util/0.24.0/opencensus-contrib-http-util-0.24.0.jar
[ERROR] urls[29] = file:/Users/anowak/.m2/repository/com/google/http-client/google-http-client-apache-v2/1.34.0/google-http-client-apache-v2-1.34.0.jar
[ERROR] urls[30] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-common-maven/1.7.0/jkube-kit-common-maven-1.7.0.jar
[ERROR] urls[31] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-common/1.7.0/jkube-kit-common-1.7.0.jar
[ERROR] urls[32] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-client/5.11.2/kubernetes-client-5.11.2.jar
[ERROR] urls[33] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-core/5.11.2/kubernetes-model-core-5.11.2.jar
[ERROR] urls[34] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-common/5.11.2/kubernetes-model-common-5.11.2.jar
[ERROR] urls[35] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-rbac/5.11.2/kubernetes-model-rbac-5.11.2.jar
[ERROR] urls[36] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-admissionregistration/5.11.2/kubernetes-model-admissionregistration-5.11.2.jar
[ERROR] urls[37] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-apps/5.11.2/kubernetes-model-apps-5.11.2.jar
[ERROR] urls[38] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-autoscaling/5.11.2/kubernetes-model-autoscaling-5.11.2.jar
[ERROR] urls[39] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-apiextensions/5.11.2/kubernetes-model-apiextensions-5.11.2.jar
[ERROR] urls[40] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-batch/5.11.2/kubernetes-model-batch-5.11.2.jar
[ERROR] urls[41] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-certificates/5.11.2/kubernetes-model-certificates-5.11.2.jar
[ERROR] urls[42] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-coordination/5.11.2/kubernetes-model-coordination-5.11.2.jar
[ERROR] urls[43] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-discovery/5.11.2/kubernetes-model-discovery-5.11.2.jar
[ERROR] urls[44] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-events/5.11.2/kubernetes-model-events-5.11.2.jar
[ERROR] urls[45] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-extensions/5.11.2/kubernetes-model-extensions-5.11.2.jar
[ERROR] urls[46] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-flowcontrol/5.11.2/kubernetes-model-flowcontrol-5.11.2.jar
[ERROR] urls[47] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-networking/5.11.2/kubernetes-model-networking-5.11.2.jar
[ERROR] urls[48] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-metrics/5.11.2/kubernetes-model-metrics-5.11.2.jar
[ERROR] urls[49] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-policy/5.11.2/kubernetes-model-policy-5.11.2.jar
[ERROR] urls[50] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-scheduling/5.11.2/kubernetes-model-scheduling-5.11.2.jar
[ERROR] urls[51] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-storageclass/5.11.2/kubernetes-model-storageclass-5.11.2.jar
[ERROR] urls[52] = file:/Users/anowak/.m2/repository/io/fabric8/kubernetes-model-node/5.11.2/kubernetes-model-node-5.11.2.jar
[ERROR] urls[53] = file:/Users/anowak/.m2/repository/com/squareup/okhttp3/okhttp/3.12.12/okhttp-3.12.12.jar
[ERROR] urls[54] = file:/Users/anowak/.m2/repository/com/squareup/okio/okio/1.15.0/okio-1.15.0.jar
[ERROR] urls[55] = file:/Users/anowak/.m2/repository/com/squareup/okhttp3/logging-interceptor/3.12.12/logging-interceptor-3.12.12.jar
[ERROR] urls[56] = file:/Users/anowak/.m2/repository/io/fabric8/zjsonpatch/0.3.0/zjsonpatch-0.3.0.jar
[ERROR] urls[57] = file:/Users/anowak/.m2/repository/com/github/mifmif/generex/1.0.2/generex-1.0.2.jar
[ERROR] urls[58] = file:/Users/anowak/.m2/repository/dk/brics/automaton/automaton/1.11-8/automaton-1.11-8.jar
[ERROR] urls[59] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-client/5.11.2/openshift-client-5.11.2.jar
[ERROR] urls[60] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-model/5.11.2/openshift-model-5.11.2.jar
[ERROR] urls[61] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-model-clusterautoscaling/5.11.2/openshift-model-clusterautoscaling-5.11.2.jar
[ERROR] urls[62] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-model-operator/5.11.2/openshift-model-operator-5.11.2.jar
[ERROR] urls[63] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-model-operatorhub/5.11.2/openshift-model-operatorhub-5.11.2.jar
[ERROR] urls[64] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-model-machine/5.11.2/openshift-model-machine-5.11.2.jar
[ERROR] urls[65] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-model-whereabouts/5.11.2/openshift-model-whereabouts-5.11.2.jar
[ERROR] urls[66] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-model-monitoring/5.11.2/openshift-model-monitoring-5.11.2.jar
[ERROR] urls[67] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-model-storageversionmigrator/5.11.2/openshift-model-storageversionmigrator-5.11.2.jar
[ERROR] urls[68] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-model-tuned/5.11.2/openshift-model-tuned-5.11.2.jar
[ERROR] urls[69] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-model-console/5.11.2/openshift-model-console-5.11.2.jar
[ERROR] urls[70] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-model-machineconfig/5.11.2/openshift-model-machineconfig-5.11.2.jar
[ERROR] urls[71] = file:/Users/anowak/.m2/repository/io/fabric8/openshift-model-miscellaneous/5.11.2/openshift-model-miscellaneous-5.11.2.jar
[ERROR] urls[72] = file:/Users/anowak/.m2/repository/org/apache/commons/commons-lang3/3.11/commons-lang3-3.11.jar
[ERROR] urls[73] = file:/Users/anowak/.m2/repository/org/apache/commons/commons-compress/1.21/commons-compress-1.21.jar
[ERROR] urls[74] = file:/Users/anowak/.m2/repository/commons-io/commons-io/2.8.0/commons-io-2.8.0.jar
[ERROR] urls[75] = file:/Users/anowak/.m2/repository/com/google/guava/guava/30.1.1-jre/guava-30.1.1-jre.jar
[ERROR] urls[76] = file:/Users/anowak/.m2/repository/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar
[ERROR] urls[77] = file:/Users/anowak/.m2/repository/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
[ERROR] urls[78] = file:/Users/anowak/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar
[ERROR] urls[79] = file:/Users/anowak/.m2/repository/org/checkerframework/checker-qual/3.8.0/checker-qual-3.8.0.jar
[ERROR] urls[80] = file:/Users/anowak/.m2/repository/com/google/errorprone/error_prone_annotations/2.5.1/error_prone_annotations-2.5.1.jar
[ERROR] urls[81] = file:/Users/anowak/.m2/repository/com/google/j2objc/j2objc-annotations/1.3/j2objc-annotations-1.3.jar
[ERROR] urls[82] = file:/Users/anowak/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.12.4/jackson-core-2.12.4.jar
[ERROR] urls[83] = file:/Users/anowak/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.12.4/jackson-databind-2.12.4.jar
[ERROR] urls[84] = file:/Users/anowak/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.12.4/jackson-annotations-2.12.4.jar
[ERROR] urls[85] = file:/Users/anowak/.m2/repository/com/fasterxml/jackson/dataformat/jackson-dataformat-yaml/2.12.4/jackson-dataformat-yaml-2.12.4.jar
[ERROR] urls[86] = file:/Users/anowak/.m2/repository/org/yaml/snakeyaml/1.27/snakeyaml-1.27.jar
[ERROR] urls[87] = file:/Users/anowak/.m2/repository/com/fasterxml/jackson/datatype/jackson-datatype-jsr310/2.12.4/jackson-datatype-jsr310-2.12.4.jar
[ERROR] urls[88] = file:/Users/anowak/.m2/repository/org/eclipse/jgit/org.eclipse.jgit/5.12.0.202106070339-r/org.eclipse.jgit-5.12.0.202106070339-r.jar
[ERROR] urls[89] = file:/Users/anowak/.m2/repository/com/googlecode/javaewah/JavaEWAH/1.1.7/JavaEWAH-1.1.7.jar
[ERROR] urls[90] = file:/Users/anowak/.m2/repository/javax/validation/validation-api/2.0.1.Final/validation-api-2.0.1.Final.jar
[ERROR] urls[91] = file:/Users/anowak/.m2/repository/org/javassist/javassist/3.20.0-GA/javassist-3.20.0-GA.jar
[ERROR] urls[92] = file:/Users/anowak/.m2/repository/com/networknt/json-schema-validator/1.0.53/json-schema-validator-1.0.53.jar
[ERROR] urls[93] = file:/Users/anowak/.m2/repository/org/apache/maven/maven-archiver/3.5.0/maven-archiver-3.5.0.jar
[ERROR] urls[94] = file:/Users/anowak/.m2/repository/org/codehaus/plexus/plexus-interpolation/1.25/plexus-interpolation-1.25.jar
[ERROR] urls[95] = file:/Users/anowak/.m2/repository/org/sonatype/plexus/plexus-sec-dispatcher/1.4/plexus-sec-dispatcher-1.4.jar
[ERROR] urls[96] = file:/Users/anowak/.m2/repository/org/sonatype/plexus/plexus-cipher/1.4/plexus-cipher-1.4.jar
[ERROR] urls[97] = file:/Users/anowak/.m2/repository/org/apache/maven/maven-builder-support/3.6.3/maven-builder-support-3.6.3.jar
[ERROR] urls[98] = file:/Users/anowak/.m2/repository/org/apache/maven/resolver/maven-resolver-util/1.4.1/maven-resolver-util-1.4.1.jar
[ERROR] urls[99] = file:/Users/anowak/.m2/repository/org/apache/maven/shared/maven-shared-utils/3.2.1/maven-shared-utils-3.2.1.jar
[ERROR] urls[100] = file:/Users/anowak/.m2/repository/javax/enterprise/cdi-api/1.0/cdi-api-1.0.jar
[ERROR] urls[101] = file:/Users/anowak/.m2/repository/javax/annotation/jsr250-api/1.0/jsr250-api-1.0.jar
[ERROR] urls[102] = file:/Users/anowak/.m2/repository/org/eclipse/sisu/org.eclipse.sisu.inject/0.3.4/org.eclipse.sisu.inject-0.3.4.jar
[ERROR] urls[103] = file:/Users/anowak/.m2/repository/com/google/inject/guice/4.2.1/guice-4.2.1-no_aop.jar
[ERROR] urls[104] = file:/Users/anowak/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar
[ERROR] urls[105] = file:/Users/anowak/.m2/repository/org/apache/maven/shared/maven-filtering/3.1.1/maven-filtering-3.1.1.jar
[ERROR] urls[106] = file:/Users/anowak/.m2/repository/org/apache/maven/plugin-tools/maven-plugin-annotations/3.6.0/maven-plugin-annotations-3.6.0.jar
[ERROR] urls[107] = file:/Users/anowak/.m2/repository/org/codehaus/plexus/plexus-archiver/3.5/plexus-archiver-3.5.jar
[ERROR] urls[108] = file:/Users/anowak/.m2/repository/org/codehaus/plexus/plexus-utils/3.0.24/plexus-utils-3.0.24.jar
[ERROR] urls[109] = file:/Users/anowak/.m2/repository/org/sonatype/plexus/plexus-build-api/0.0.7/plexus-build-api-0.0.7.jar
[ERROR] urls[110] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-enricher-generic/1.7.0/jkube-kit-enricher-generic-1.7.0.jar
[ERROR] urls[111] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-enricher-api/1.7.0/jkube-kit-enricher-api-1.7.0.jar
[ERROR] urls[112] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-config-image/1.7.0/jkube-kit-config-image-1.7.0.jar
[ERROR] urls[113] = file:/Users/anowak/.m2/repository/io/fabric8/ianaservicehelper/0.0.1/ianaservicehelper-0.0.1.jar
[ERROR] urls[114] = file:/Users/anowak/.m2/repository/com/google/code/gson/gson/2.8.6/gson-2.8.6.jar
[ERROR] urls[115] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-enricher-specific/1.7.0/jkube-kit-enricher-specific-1.7.0.jar
[ERROR] urls[116] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-resource-service/1.7.0/jkube-kit-resource-service-1.7.0.jar
[ERROR] urls[117] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-resource-helm/1.7.0/jkube-kit-resource-helm-1.7.0.jar
[ERROR] urls[118] = file:/Users/anowak/.m2/repository/org/apache/httpcomponents/httpclient/4.5.13/httpclient-4.5.13.jar
[ERROR] urls[119] = file:/Users/anowak/.m2/repository/org/apache/httpcomponents/httpcore/4.4.13/httpcore-4.4.13.jar
[ERROR] urls[120] = file:/Users/anowak/.m2/repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar
[ERROR] urls[121] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-generator-java-exec/1.7.0/jkube-kit-generator-java-exec-1.7.0.jar
[ERROR] urls[122] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-generator-api/1.7.0/jkube-kit-generator-api-1.7.0.jar
[ERROR] urls[123] = file:/Users/anowak/.m2/repository/org/jboss/shrinkwrap/shrinkwrap-api/1.2.6/shrinkwrap-api-1.2.6.jar
[ERROR] urls[124] = file:/Users/anowak/.m2/repository/org/jboss/shrinkwrap/shrinkwrap-spi/1.2.6/shrinkwrap-spi-1.2.6.jar
[ERROR] urls[125] = file:/Users/anowak/.m2/repository/org/jboss/shrinkwrap/shrinkwrap-impl-base/1.2.6/shrinkwrap-impl-base-1.2.6.jar
[ERROR] urls[126] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-generator-karaf/1.7.0/jkube-kit-generator-karaf-1.7.0.jar
[ERROR] urls[127] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-generator-webapp/1.7.0/jkube-kit-generator-webapp-1.7.0.jar
[ERROR] urls[128] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-thorntail/1.7.0/jkube-kit-thorntail-1.7.0.jar
[ERROR] urls[129] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-spring-boot/1.7.0/jkube-kit-spring-boot-1.7.0.jar
[ERROR] urls[130] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-watcher-api/1.7.0/jkube-kit-watcher-api-1.7.0.jar
[ERROR] urls[131] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-quarkus/1.7.0/jkube-kit-quarkus-1.7.0.jar
[ERROR] urls[132] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-micronaut/1.7.0/jkube-kit-micronaut-1.7.0.jar
[ERROR] urls[133] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-openliberty/1.7.0/jkube-kit-openliberty-1.7.0.jar
[ERROR] urls[134] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-vertx/1.7.0/jkube-kit-vertx-1.7.0.jar
[ERROR] urls[135] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-wildfly-jar/1.7.0/jkube-kit-wildfly-jar-1.7.0.jar
[ERROR] urls[136] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-profiles/1.7.0/jkube-kit-profiles-1.7.0.jar
[ERROR] urls[137] = file:/Users/anowak/.m2/repository/org/eclipse/jkube/jkube-kit-watcher-standard/1.7.0/jkube-kit-watcher-standard-1.7.0.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------
[ERROR] : ExceptionInInitializerError: Can't overwrite cause with java.lang.UnsatisfiedLinkError: java.lang.UnsatisfiedLinkError: Can't load library: /var/folders/8q/b1g02zn17qb5_grv0pcj5xc00000gn/T/jffi279835088202327901.dylib
[ERROR]         at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2398)
[ERROR]         at java.base/java.lang.Runtime.load0(Runtime.java:785)
[ERROR]         at java.base/java.lang.System.load(System.java:1979)
[ERROR]         at com.kenai.jffi.internal.StubLoader.loadFromJar(StubLoader.java:361)
[ERROR]         at com.kenai.jffi.internal.StubLoader.load(StubLoader.java:254)
[ERROR]         at com.kenai.jffi.internal.StubLoader.<clinit>(StubLoader.java:434)
[ERROR]         at java.base/java.lang.Class.forName0(Native Method)
[ERROR]         at java.base/java.lang.Class.forName(Class.java:488)
[ERROR]         at java.base/java.lang.Class.forName(Class.java:467)
[ERROR]         at com.kenai.jffi.Init.load(Init.java:68)
[ERROR]         at com.kenai.jffi.Foreign$InstanceHolder.getInstanceHolder(Foreign.java:49)
[ERROR]         at com.kenai.jffi.Foreign$InstanceHolder.<clinit>(Foreign.java:45)
[ERROR]         at com.kenai.jffi.Foreign.getInstance(Foreign.java:103)
[ERROR]         at com.kenai.jffi.Type$Builtin.lookupTypeInfo(Type.java:242)
[ERROR]         at com.kenai.jffi.Type$Builtin.getTypeInfo(Type.java:237)
[ERROR]         at com.kenai.jffi.Type.resolveSize(Type.java:155)
[ERROR]         at com.kenai.jffi.Type.size(Type.java:138)
[ERROR]         at jnr.ffi.provider.jffi.NativeRuntime$TypeDelegate.size(NativeRuntime.java:178)
[ERROR]         at jnr.ffi.provider.AbstractRuntime.<init>(AbstractRuntime.java:48)
[ERROR]         at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:57)
[ERROR]         at jnr.ffi.provider.jffi.NativeRuntime.<init>(NativeRuntime.java:41)
[ERROR]         at jnr.ffi.provider.jffi.NativeRuntime$SingletonHolder.<clinit>(NativeRuntime.java:53)
[ERROR]         at jnr.ffi.provider.jffi.NativeRuntime.getInstance(NativeRuntime.java:49)
[ERROR]         at jnr.ffi.provider.jffi.Provider.<init>(Provider.java:29)
[ERROR]         at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:67)
[ERROR]         at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
[ERROR]         at java.base/java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:128)
[ERROR]         at java.base/jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:341)
[ERROR]         at java.base/java.lang.Class.newInstance(Class.java:677)
[ERROR]         at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.getInstance(FFIProvider.java:68)
[ERROR]         at jnr.ffi.provider.FFIProvider$SystemProviderSingletonHolder.<clinit>(FFIProvider.java:57)
[ERROR]         at jnr.ffi.provider.FFIProvider.getSystemProvider(FFIProvider.java:35)
[ERROR]         at jnr.ffi.LibraryLoader.create(LibraryLoader.java:73)
[ERROR]         at jnr.unixsocket.Native.<clinit>(Native.java:72)
[ERROR]         at jnr.unixsocket.UnixSocketChannel.<init>(UnixSocketChannel.java:94)
[ERROR]         at jnr.unixsocket.UnixSocketChannel.open(UnixSocketChannel.java:49)
[ERROR]         at org.eclipse.jkube.kit.build.service.docker.helper.LocalSocketUtil.canConnectUnixSocket(LocalSocketUtil.java:39)
[ERROR]         at org.eclipse.jkube.kit.build.service.docker.access.DockerConnectionDetector$UnixSocketDockerHostProvider.getConnectionParameter(DockerConnectionDetector.java:130)
[ERROR]         at org.eclipse.jkube.kit.build.service.docker.access.DockerConnectionDetector.detectConnectionParameter(DockerConnectionDetector.java:98)
[ERROR]         at org.eclipse.jkube.kit.build.service.docker.DockerAccessFactory.createDockerAccess(DockerAccessFactory.java:43)
[ERROR]         at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.doExecute(AbstractDockerMojo.java:449)
[ERROR]         at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.execute(AbstractDockerMojo.java:423)
[ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:301)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:211)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:157)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:121)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR]         at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:127)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:294)
[ERROR]         at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR]         at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR]         at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)
[ERROR]         at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
[ERROR]         at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
[ERROR]         at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:577)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR]         at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginContainerException
</pre>
</details>

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -52,6 +52,7 @@ Usage:
 * Fix #1382: Docker Build ARGS not replaced properly
 * Fix #1324: Support legacy javaee as well as jakartaee projects in the Tomcat webapp generator
 * Fix #1460: Route doesn't use the service "normalizePort"
+* Fix #1470: Add support for Apple M1 CPUs
 * Fix #1482: Quarkus Generator and Enricher should be compatible with the Red Hat build
 * Fix #1483: Assembly files with unnormalized paths don't get fileMode changes
 * Fix #1489: Align BaseGenerator's `add` and `tags` config options to work with `jkube.generator.*` properties

--- a/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/access/hc/unix/UnixSocket.java
+++ b/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/access/hc/unix/UnixSocket.java
@@ -159,13 +159,13 @@ final class UnixSocket extends Socket {
     }
 
     @Override
-    public synchronized void setSoTimeout(int timeout) {
-        channel.setSoTimeout(timeout);
+    public synchronized void setSoTimeout(int timeout) throws SocketException {
+        channel.socket().setSoTimeout(timeout);
     }
 
     @Override
-    public synchronized int getSoTimeout() {
-        return channel.getSoTimeout();
+    public synchronized int getSoTimeout() throws SocketException {
+        return channel.socket().getSoTimeout();
     }
 
     @Override
@@ -213,13 +213,13 @@ final class UnixSocket extends Socket {
     }
 
     @Override
-    public void setKeepAlive(boolean on) {
-        channel.setKeepAlive(on);
+    public void setKeepAlive(boolean on) throws SocketException {
+        channel.socket().setKeepAlive(on);
     }
 
     @Override
-    public boolean getKeepAlive() {
-        return channel.getKeepAlive();
+    public boolean getKeepAlive() throws SocketException {
+        return channel.socket().getKeepAlive();
     }
 
     @Override

--- a/jkube-kit/build/service/docker/src/test/java/org/eclipse/jkube/kit/build/service/docker/access/hc/unix/UnixSocketTest.java
+++ b/jkube-kit/build/service/docker/src/test/java/org/eclipse/jkube/kit/build/service/docker/access/hc/unix/UnixSocketTest.java
@@ -1,0 +1,92 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.build.service.docker.access.hc.unix;
+
+import jnr.unixsocket.UnixSocketChannel;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.MockedStatic;
+
+import java.io.IOException;
+import java.net.SocketException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.mockito.Mockito.*;
+
+public class UnixSocketTest {
+    private static final int SO_TIMEOUT = 60;
+    private static final boolean KEEP_ALIVE = true;
+
+    private MockedStatic<UnixSocketChannel> unixSocketChannel;
+    private UnixSocketChannel socketChannel;
+    private jnr.unixsocket.UnixSocket socket;
+
+    @Before
+    public void setUp() {
+        socketChannel = mock(UnixSocketChannel.class);
+        socket = mock(jnr.unixsocket.UnixSocket.class);
+        doReturn(socket).when(socketChannel).socket();
+        unixSocketChannel = mockStatic(UnixSocketChannel.class);
+        unixSocketChannel.when(() -> UnixSocketChannel.open()).thenReturn(socketChannel);
+    }
+
+    @After
+    public void close() {
+        unixSocketChannel.close();
+    }
+
+    @Test
+    public void shouldReturnValuesFromSocket() throws IOException {
+        // GIVEN
+        UnixSocket unixSocket = new UnixSocket();
+        doReturn(KEEP_ALIVE).when(socket).getKeepAlive();
+        doReturn(SO_TIMEOUT).when(socket).getSoTimeout();
+
+        // WHEN
+        boolean keepAlive = unixSocket.getKeepAlive();
+        int soTimeout = unixSocket.getSoTimeout();
+
+        // THEN
+        assertThat(keepAlive).isEqualTo(KEEP_ALIVE);
+        assertThat(soTimeout).isEqualTo(SO_TIMEOUT);
+    }
+
+    @Test
+    public void shouldPassExceptionsFromSocket() throws IOException {
+        // GIVEN
+        UnixSocket unixSocket = new UnixSocket();
+        doThrow(new SocketException()).when(socket).getKeepAlive();
+        doThrow(new SocketException()).when(socket).getSoTimeout();
+
+        // WHEN & THEN
+        assertThatThrownBy(() -> unixSocket.getKeepAlive()).isInstanceOf(SocketException.class);
+        assertThatThrownBy(() -> unixSocket.getSoTimeout()).isInstanceOf(SocketException.class);
+    }
+
+    @Test
+    public void shouldForwardValuesToSocket() throws IOException {
+        // GIVEN
+        UnixSocket unixSocket = new UnixSocket();
+
+        // WHEN
+        unixSocket.setKeepAlive(KEEP_ALIVE);
+        unixSocket.setSoTimeout(SO_TIMEOUT);
+
+        // THEN
+        verify(socket).setKeepAlive(KEEP_ALIVE);
+        verify(socket).setSoTimeout(SO_TIMEOUT);
+    }
+}

--- a/jkube-kit/build/service/docker/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker
+++ b/jkube-kit/build/service/docker/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker
@@ -1,0 +1,1 @@
+mock-maker-inline

--- a/jkube-kit/parent/pom.xml
+++ b/jkube-kit/parent/pom.xml
@@ -46,7 +46,7 @@
     <version.jansi>2.3.3</version.jansi>
     <version.javassist>3.20.0-GA</version.javassist>
     <version.jgit>5.12.0.202106070339-r</version.jgit>
-    <version.jnr-unixsocket>0.12</version.jnr-unixsocket>
+    <version.jnr-unixsocket>0.38.17</version.jnr-unixsocket>
     <version.json-smart>2.2.1</version.json-smart> <!-- Transitive dependency from com.consol.citrus:citrus-core -->
     <version.jsr305>3.0.2</version.jsr305>
     <version.JUnitParams>1.1.1</version.JUnitParams>
@@ -66,6 +66,7 @@
     <version.shrinkwrap>1.2.6</version.shrinkwrap>
     <version.validation-api>2.0.1.Final</version.validation-api>
     <version.wagon-ssh-external>2.3</version.wagon-ssh-external>
+    <version.ow2.asm>5.0.3</version.ow2.asm>
 
     <!-- =======================================================  -->
     <version.image.jkube-images>0.0.15</version.image.jkube-images>
@@ -558,6 +559,12 @@
         <groupId>org.slf4j</groupId>
         <artifactId>slf4j-api</artifactId>
         <version>${version.slf4j-api}</version>
+      </dependency>
+
+      <dependency>
+        <groupId>org.ow2.asm</groupId>
+        <artifactId>asm</artifactId>
+        <version>${version.ow2.asm}</version>
       </dependency>
 
       <!-- == maven ===================================== -->

